src/rust/grapl-utils/src/future_ext/mod.rs

```diff
@@ -25,3 +25,32 @@ pub trait GraplFutureExt: Future {
 }
 
 impl<T: ?Sized> GraplFutureExt for T where T: Future {}
+
+#[async_trait::async_trait]
+pub trait GraplAsyncRetry<Fut, T, E>: Fn() -> Fut
+where
+    Fut: Future<Output = Result<T, E>> + Send,
+    E: std::error::Error + Send + Sync + 'static,
+{
+    /// Helper method to auto-retry an async () -> Result.
+    /// If it fails after `num_retries + 1` times, return the Err.
+    async fn retry(&self, num_retries: u8) -> Result<T, E> {
```

Comment:
fwiw elsewhere we use `futures-retry` for this sort of thing (graph-merger), but nbd

Reply:
done